R/data_reshape.R

```diff
@@ -175,66 +175,131 @@ data_to_long <- function(data,
 #' @export
 data_to_wide <- function(data,
                          values_from = "Value",
-                         colnames_from = "Name",
-                         rows_from = NULL,
-                         sep = "_",
+                         names_from = "Name",
+                         names_sep = "_",
+                         names_prefix = "",
+                         values_fill = NULL,
+                         verbose = TRUE,
                          ...,
-                         names_from = colnames_from,
-                         verbose = TRUE) {
+                         colnames_from,
+                         sep) {
+
+  .is_deprecated(colnames_from, "names_from")
+  .is_deprecated(sep, "names_sep")
+
+  if (!missing(colnames_from) && is.null(names_from)) names_from <- colnames_from
+  if (!missing(sep) && is.null(names_sep)) names_sep <- sep
+
+  old_names <- names(data)
+
+  # Preserve attributes
   if (inherits(data, "tbl_df")) {
     tbl_input <- TRUE
     data <- as.data.frame(data)
   } else {
     tbl_input <- FALSE
   }
-
-  # Compatibility with tidyr
-  if (names_from != colnames_from) colnames_from <- names_from
-
-  # save attribute of each variable
   variable_attr <- lapply(data, attributes)
 
-  # If no other row identifier, create one
-  if (is.null(rows_from)) {
-    if (all(names(data) %in% c(values_from, colnames_from))) {
-      data[["_Rows"]] <- row.names(data)
-    }
-    data[["_Rows"]] <- apply(data[, !names(data) %in% c(values_from, colnames_from), drop = FALSE], 1, paste, collapse = "_")
-    rows_from <- "_Rows"
+
+  # Create an id for stats::reshape
+  if (all(names(data) %in% c(values_from, names_from))) {
+    data[["_Rows"]] <- row.names(data)
+  } else {
+    data[["_Rows"]] <- apply(data[, !names(data) %in% c(values_from, names_from), drop = FALSE], 1, paste, collapse = "_")
   }
+  rows_from <- "_Rows"
+
 
   # create pattern of column names - stats::reshape renames columns that
   # concatenates "v.names" + values - we only want values
-  old_colnames <- paste0(values_from, "_", unique(data[[colnames_from]]))
-  new_colnames <- unique(data[[colnames_from]])
+  current_colnames <- colnames(data)
+  current_colnames <- current_colnames[current_colnames != "_Rows"]
+  future_colnames <- unique(apply(data, 1, function(x) paste(x[c(names_from)], collapse = names_sep)))
+
+  # stop if some column names would be duplicated (follow tidyr workflow)
+  if (any(future_colnames %in% current_colnames)) {
+    stop(insight::format_message(
+      "Some values of the columns specified in 'names_from' are already present as column names.",
+      paste0("Either use `name_prefix` or rename the following columns: ",
```

Comment:
I just discovered this _datawizdard_(!) function just recently, but I suggest using `text_concatenate()`:

``` r
current_colnames <- colnames(iris)
future_colnames <- current_colnames[2:4]

paste0("Either use `name_prefix` or rename the following columns: ",
       paste(current_colnames[which(current_colnames %in% future_colnames)],
             collapse = ", "))
#> [1] "Either use `name_prefix` or rename the following columns: Sepal.Width, Petal.Length, Petal.Width"

paste0("Either use `name_prefix` or rename the following columns: ", 
       datawizard::text_concatenate(current_colnames[which(current_colnames %in% future_colnames)],
                                    enclose = "`"))
#> [1] "Either use `name_prefix` or rename the following columns: `Sepal.Width`, `Petal.Length` and `Petal.Width`"
```

<sup>Created on 2022-06-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
(you may omit the `enclose` argument)

Reply:
done (but not sure it simplifies the code in this case)